plugin/evm/warp/backend.go

```diff
@@ -46,17 +46,16 @@ func NewWarpBackend(snowCtx *snow.Context, db database.Database, signatureCacheS
 func (w *warpBackend) AddMessage(ctx context.Context, unsignedMessage *teleporter.UnsignedMessage) error {
 	messageID := hashing.ComputeHash256Array(unsignedMessage.Bytes())
 
-	// We generate the signature here and only save the signature in the db and cache.
-	// It is left to smart contracts built on top of Warp to save messages if required.
+	// We save the message instead of signature for db, in case for bls key changes.
```

Comment:
if bls keys may change, (when) is it OK to cache the signature?

Reply:
This is related to [this comment](https://github.com/ava-labs/subnet-evm/pull/452#discussion_r1084669771). Signatures should be okay to cache while normally running for adding messages and getting signatures, but when the node goes offline and possibly changes its bls key, the cache would be wiped. Then after restart, the cache gets repopulated with signatures with the updated bls key.

Reply:
yes, the assumption is that the BLS key will only change after a restart.

Reply:
could we capture this in a comment in code? otherwise, LGTM